Dockerfile

```diff
@@ -272,63 +274,52 @@ RUN chroot "$ROOTFS" prltoolsd -V
+	&& cd /linux-kernel/tools/hv \
+	&& sed -i 's!\(^CFLAGS = .*\)!\1 -I/tmp/kheaders/include!' Makefile \
+	&& make hv_kvp_daemon \
+	&& cp hv_kvp_daemon $ROOTFS/usr/sbin \
+	&& rm -rf /tmp/kheaders
+
 # Make sure that all the modules we might have added are recognized (especially VBox guest additions)
-RUN depmod -a -b $ROOTFS $KERNEL_VERSION-boot2docker
+RUN depmod -a -b "$ROOTFS" "$KERNEL_VERSION-boot2docker"
 
 COPY VERSION $ROOTFS/etc/version
-RUN cp -v $ROOTFS/etc/version /tmp/iso/version
+RUN cp -v "$ROOTFS/etc/version" /tmp/iso/version
 
 # Get the Docker binaries with version that matches our boot2docker version.
-RUN curl -fSL -o /tmp/dockerbin.tgz https://get.docker.com/builds/Linux/x86_64/docker-$(cat $ROOTFS/etc/version).tgz && \
-    tar -zxvf /tmp/dockerbin.tgz -C "$ROOTFS/usr/local/bin" --strip-components=1 && \
-    rm /tmp/dockerbin.tgz && \
-    chroot "$ROOTFS" docker -v
+RUN set -ex \
+	&& curl -fSL -o /tmp/dockerbin.tgz "https://get.docker.com/builds/Linux/x86_64/docker-$(cat "$ROOTFS/etc/version").tgz" \
+	&& tar -zxvf /tmp/dockerbin.tgz -C "$ROOTFS/usr/local/bin" --strip-components=1 \
+	&& rm /tmp/dockerbin.tgz \
+	&& chroot "$ROOTFS" docker -v
 
 # Copy our custom rootfs
 COPY rootfs/rootfs $ROOTFS
 
 # setup acpi config dir &
 # tcl6's sshd is compiled without `/usr/local/sbin` in the path
-# Boot2Docker and Docker Machine need `ip`, so I'm linking it in here
-RUN cd $ROOTFS \
-    && ln -s /usr/local/etc/acpi etc/ \
-    && ln -s /usr/local/sbin/ip usr/sbin/
+# Boot2Docker and Docker Machine need `ip`, so link it elsewhere
+RUN ln -svT /usr/local/etc/acpi "$ROOTFS/etc/acpi" \
+	&& ln -svT /usr/local/sbin/ip "$ROOTFS/usr/sbin/ip"
 
-# Build the Hyper-V KVP Daemon
-RUN cd /linux-kernel && \
-    make INSTALL_HDR_PATH=/tmp/kheaders headers_install && \
-    cd /linux-kernel/tools/hv && \
-    sed -i 's!\(^CFLAGS = .*\)!\1 -I/tmp/kheaders/include!' Makefile && \
-    make hv_kvp_daemon && \
-    cp hv_kvp_daemon $ROOTFS/usr/sbin && \
-    rm -rf /tmp/kheaders
-
-# These steps can only be run once, so can't be in make_iso.sh (which can be run in chained Dockerfiles)
+# These steps should only be run once, so can't be in make_iso.sh (which can be run in chained Dockerfiles)
 # see https://github.com/boot2docker/boot2docker/blob/master/doc/BUILD.md
 
 # Make sure init scripts are executable
-RUN find $ROOTFS/etc/rc.d/ $ROOTFS/usr/local/etc/init.d/ -exec chmod +x '{}' ';'
+RUN find "$ROOTFS/etc/rc.d/" "$ROOTFS/usr/local/etc/init.d/" -type f -exec chmod --changes +x '{}' +
 
 # move dhcp.sh out of init.d as we're triggering it manually so its ready a bit faster
-RUN mv $ROOTFS/etc/init.d/dhcp.sh $ROOTFS/etc/rc.d/
-
-# Change MOTD
-RUN mv $ROOTFS/usr/local/etc/motd $ROOTFS/etc/motd
-
-# Make sure we have the correct bootsync
-RUN mv $ROOTFS/boot*.sh $ROOTFS/opt/ && \
-	chmod +x $ROOTFS/opt/*.sh
-
-# Make sure we have the correct shutdown
-RUN mv $ROOTFS/shutdown.sh $ROOTFS/opt/shutdown.sh && \
-	chmod +x $ROOTFS/opt/shutdown.sh
```

Comment:
(except `dhcp.sh`, which comes from the TCL rootfs and we move it into `/etc/rc.d` and invoke it directly ourselves)